echopype/convert/set_groups_ek80.py

```diff
@@ -660,12 +659,51 @@ def _assemble_ds_power(self, ch):
                 ),
                 "range_sample": (
                     ["range_sample"],
-                    np.arange(data_shape[1]),
+                    np.arange(self.parser_obj.ping_data_dict["power"][ch].shape[1]),
                     self._varattrs["beam_coord_default"]["range_sample"],
                 ),
             },
         )
 
+        # If RAW4 datagram (transmit pulse in complex) exists
+        if len(self.parser_obj.ping_data_dict_tx["complex"]) != 0:
+            # Add coordinate transmit_sample
+            ds_tmp = ds_tmp.assign_coords(
+                {
+                    "transmit_sample": (
+                        ["transmit_sample"],
```

Comment:
That is a very good point!  I'll add them!

Reply:
Actually, it's already done, because this section is within `_assemble_ds_power` that assembles a dataset for a given channel. All channel-specific datasets are then expanded with the dimension `channel` and then merged into a single one at the end
https://github.com/OSOceanAcoustics/echopype/blob/ad15001d29d64cac769a686132a1d9bd6a902748/echopype/convert/set_groups_ek80.py#L854-L857
If you check the final dataset, these variables have a `channel` dimension.

I've added checking for the dimensions `('channel', 'ping_time', 'transmit_sample')` to be part of the test though, to be explicit on what we expect there.